include/circt/Dialect/FIRRTL/FIRRTLDialect.h

```diff
@@ -22,10 +22,18 @@ namespace firrtl {
 
 class FIRRTLType;
 
-/// If the specified attribute list has a firrtl.name attribute, return its
+/// If the specified module has an argnames attribute, return its
 /// value.
-StringAttr getFIRRTLNameAttr(ArrayRef<NamedAttribute> attrs);
+StringAttr getFIRRTLModuleArgNameAttr(Operation *module, size_t argNo);
 
+bool hasFIRRTLModuleArgNameAttr(Operation *module);
+
+static inline StringRef getFIRRTLModuleArgName(Operation *module,
+                                               size_t argNo) {
+  return hasFIRRTLModuleArgNameAttr(module)
+             ? getFIRRTLModuleArgNameAttr(module, argNo).getValue()
+             : "";
```

Comment:
Is it safe to return `""` as a `StringRef`? An alternative formulation of this would be to guarantee that this can succeed (there exists a `StringAttr`) and return the ref to that.

Reply:
Also, the change to module construction to pre-populate everything with `StringAttr("")` for empty names should mean that false path can never happen?